modules/platforms/dotnet/Apache.Ignite.Core.Tests/Common/IgniteProductVersionTests.cs

```diff
@@ -0,0 +1,158 @@
+
+            IgniteProductVersion deserializedVersion = new IgniteProductVersion(reader);
+
+            Assert.AreEqual(_defaultVersion, deserializedVersion);
+        }
+
+        private IBinaryRawReader SetUpRawBinaryReader(byte[] hash)
+        {
+            var marsh = BinaryUtils.Marshaller;
+
+            using (var stream = new BinaryHeapStream(128))
+            {
+                var writer = marsh.StartMarshal(stream);
+
+                writer.WriteByte(_defaultVersion.Major);
+                writer.WriteByte(_defaultVersion.Minor);
+                writer.WriteByte(_defaultVersion.Maintenance);
+                writer.WriteLong(_defaultVersion.RevisionTimestamp);
+                writer.WriteByteArray(hash);
+
+                stream.Seek(0, SeekOrigin.Begin);
+
+                return marsh.StartUnmarshal(stream);
+            }
+        }
+
+        [Test]
+        public void TestEqualityWithNull()
+        {
+            Assert.IsFalse(_defaultVersion.Equals(null));
+        }
+
+        [Test]
+        public void TestToString()
+        {
+            Assert.AreEqual("2.5.7#20180101-sha1:", _defaultVersion.ToString());
+        }
+
+        [Test]
+        public void TestToStringWithLimitedHashValue()
+        {
+            var hash = new byte[] {24, 229};
+            IBinaryRawReader reader = SetUpRawBinaryReader(hash);
+
+            IgniteProductVersion deserializedVersion = new IgniteProductVersion(reader);
+            Assert.AreEqual("2.5.7#20180101-sha1:18", deserializedVersion.ToString());
+        }
+
+        [Test]
+        public void TestLocalNodeGetVersion()
+        {
+            using (IIgnite ignite = Ignition.Start(TestUtils.GetTestConfiguration()))
+            {
+                IgniteProductVersion nodeVersion = ignite.GetCluster().GetLocalNode().Version;
+
+                Assert.GreaterOrEqual(nodeVersion.Major, 2);
+                Assert.GreaterOrEqual(nodeVersion.Minor, 0);
+                Assert.GreaterOrEqual(nodeVersion.Maintenance, 0);
+            }
+        }
+
+        [Test]
+        public void TestIgniteGetVersionAndNodeVersionAreEqual()
+        {
+            using (IIgnite ignite = Ignition.Start(TestUtils.GetTestConfiguration()))
+            {
+                IgniteProductVersion version = ignite.GetVersion();
+                IgniteProductVersion nodeVersion = ignite.GetCluster().GetLocalNode().Version;
+
+                Assert.AreEqual(version, nodeVersion);
+            }
+        }
+
+        [Test]
+        public void TestClientVersionIsMoreOrEqualsServerNodeVersion()
+        {
+            using (IIgnite ignite = Ignition.Start(TestUtils.GetTestConfiguration()))
+            {
+                IgniteProductVersion version = ignite.GetVersion();
+
+                Version clientVer = Assembly.GetExecutingAssembly().GetName().Version;
+                var expectedVersion = new IgniteProductVersion((byte) clientVer.Major, (byte) clientVer.Minor,
```

Comment:
Looks misleading.
Let's compare only major, minor, and patch for strict equality as three asserts.